Python/context.c

```diff
@@ -115,7 +115,9 @@ context_event_name(PyContextEvent event) {
 static void notify_context_watchers(PyContextEvent event, PyContext *ctx)
 {
     assert(Py_REFCNT(ctx) > 0);
-    PyInterpreterState *interp = _PyInterpreterState_GET();
+    PyThreadState *ts = _PyThreadState_GET();
```

Comment:
You can pass tstate from the callee, `_PyContext_Enter`

Reply:
sorry, completely missed that been months since I looked at this.